python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -153,6 +155,107 @@ def update_cursor(self, cursor: Optional[str]) -> None:
     def repository_name(self) -> Optional[str]:
         return self._repository_name
 
+class MultiAssetSensorEvaluationContext(SensorEvaluationContext):
+    """The context object available as the argument to the evaluation function of a :py:class:`dagster.MultiAssetSensorDefinition`.
+
+    Users should not instantiate this object directly. To construct a
+    `MultiAssetSensorEvaluationContext` for testing purposes, use :py:func:`dagster.
+    build_multi_asset_sensor_context`.
+
+    Attributes:
+        instance_ref (Optional[InstanceRef]): The serialized instance configured to run the schedule
+        cursor (Optional[str]): The cursor, passed back from the last sensor evaluation via
+            the cursor attribute of SkipReason and RunRequest
+        last_completion_time (float): DEPRECATED The last time that the sensor was evaluated (UTC).
+        last_run_key (str): DEPRECATED The run key of the RunRequest most recently created by this
+            sensor. Use the preferred `cursor` attribute instead.
+        repository_name (Optional[str]): The name of the repository that the sensor belongs to.
+        instance (Optional[DagsterInstance]): The deserialized instance can also be passed in
+            directly (primarily useful in testing contexts).
+
+    Example:
+
+    .. code-block:: python
+
+        from dagster import sensor, SensorEvaluationContext
+
+        @sensor
+        def the_sensor(context: SensorEvaluationContext):
+            ...
+
+    """
+
+    def get_latest_events(self, asset_keys: Sequence[AssetKey]):
```

Comment:
I think a name like `latest_materializations_by_key` would better help users understand what this is going to return. Also, I would expect the keys of the dictionary to be AssetKeys instead of strings.

Reply:
Also, can we use the asset keys from the sensor for this instead of asking the user to provide them? Requiring users to provide the asset keys would solve the ambiguity I brought up in my last review about which keys are included, but IMO it's not worth the additional boilerplate for users.

Reply:
i like that name, and yeah the keys being asset keys makes more sense.  

I'm thinking the function could take an optional list of keys and if none are provided it'll default to the list provided to the sensor (just didn't want to implement rn in case we scrapped the new approach)

Reply:
Accepting an optional list of keys sounds sensible to me